### ISSUE TITLE
onpc connection issue while running Wireguard

### ISSUE THREAD

**Issue by avuton** [mkulesh/onpc#192]
Hello! First, let me thank you for the marvelous work. I'm a happy daily user.

Let me start by saying that I don't believe this issue has anything to do with onpc. I'm at my wits end at troubleshooting and I'm hoping a kind soul can shove me in the right direction.

Issue: When connected to my Onkyo TX-NR656 on either zone, with Wireguard on, within 10 seconds the connection will disconnect.

Known:
* This issue was not present for the last several months up until recently when the issue began (2 weeks ago?).
* This issue is present on both my OnePlus 5T on LineageOS and a LG Tablet (newer Android, don't recall the version); the APK for both are from F-Droid; I have tried from the Google Play Store and built in Android Studio.
* This does not occur with Wireguard active and Enhanced Controller for iOS.
* This receiver just received a firmware update soon before this failure began, if I recall correctly. I am not aware of any way to downgrade the firmware, though I do have an old version on hand if someone could clue me in. [This](http://divideoverflow.com/2015/05/how-to-update-onkyo-firmware-to-older-version) does not work; there is no USB button on this receiver.
* Turn off Wireguard and the onpc connection is stable.
* Downgrade of Wireguard on the server and the app to old, known working, versions do not fix the issue.
* Downgrade of onpc to old versions don't fix the issue.
* Strangely enough, no other applications are affected in a similar way that I am aware of. [M.A.L.P.](https://play.google.com/store/apps/details?id=org.gateshipone.malp&hl=en_US&gl=US), for example, continues chugging along just fine.
* Power cycling the receiver doesn't help.
* There is no connection error if I netcat from the OnePlus 5T to the receiver when sending no commands with Wireguard activated.

Here's an example logcat with Logging.isEnable = true and Logging.isTimeMessageEnabled = true: https://gist.github.com/avuton/87279ca789b4f692a59096be9f9433b5

**Comment by mkulesh** [comment]
Hi @avuton,
unfortunately the log on gist.github.com is truncated. Could you please send me whole log on mikhail.kulesh@gmail.com?
Do you have the log from Wireguard? Interesting is does Wireguard log something when the app is disconnected? 
As I see on https://www.intl.onkyo.com/support/firmware/tx-nr656.html, there are no network issues fixed in the latest firmware update... May be "Improve the stability of operation when returning from Network Standby"? That happens if you first power off the receiver manually, wait more than 10 seconds and than connect receiver from the Android App? And if you already have Android Studio installed... Could you please create any Android emulator in Android Studio (Tools -> AVD manager), try to run the app in emulator and connect the receiver from it?

**Comment by mkulesh** [comment]
Hi @avuton,
an other experiment... Could you try to install https://github.com/mkulesh/onpc/raw/autobuild/autobuild/onpc-v2.7-release.apk
It is an Android build of the Enhanced Controller for iOS. Does this build also have this problem?

**Comment by avuton** [comment]
Unfortunately, yes.

**Comment by mkulesh** [comment]
"Unfortunately, yes" do you mean that onpc-v2.7-release.apk also disconnects after 10 seconds?

**Comment by avuton** [comment]
Yes, sorry I wasn't clear, it also disconnects. [here's a screencap](https://nc.anpmech.com/index.php/f/4542403)

**Comment by avuton** [comment]
Sorry, I was unclear, yes it also happens with that build. Here's a [screencap](https://nc.anpmech.com/index.php/s/FGDHFxqEqz76Lzw).

**Comment by mkulesh** [comment]
Do you have the log from Wireguard? That logs Wireguard at the moment when app is disconnected?

**Comment by avuton** [comment]
I am so sorry, I was wrong :( I downgraded kernels, but didn't think about the fact that the Wireguard module being built was the same version that I was using (it had to be built via DKMS since it was a super old kernel) and, apparently, this Wireguard was buggy. I since updated and the problem has gone away. Thank you for the help, and I truly apologize that it was a bit of a waste of time. I have no idea why this app was targeted with that bug, but I'm definitely relieved to be using it again without connection issues.
